# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [17]:
import pandas as pd
import numpy as np
from scipy import stats
pd.set_option('display.max_columns', None)
import math

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

The question here is find out if there is a difference in the average call rate between White race applicants and other race applicants. So, we are comparing the mean of two samples, $X bar$, where the population standard deviation σ is not known. The sample size, n, is > 30. Considering all these, we will use a two sample $t$ test to solve this. Yes, Central Limit Theorem does apply.

Ho : Xbarwhite pop = Xbarblack pop
Ha : Xbarwhite pop ≠ Xbarblack pop

In [5]:
data_b=data[data.race=='b']

In [6]:
data_b.head(5)

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
7  b  1          3       4        21       0          1         0         1   
8  b  1          4       3         3       0          0         0         0   
9  b  1          4       2         6       0          1         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
7            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   
8            316    ...          0.0     0.0    0.0       0.0       0.0   1.0   
9            263    ...          0.0     0.0    0.0       0.0       0.0   1.0   

  busservice othservice  missind  ownership  
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
7        0.0        1.0      0.0  Nonprofit  
8        0.0        0.0      0.0    Private  
9        0.0        0.0      0.0    Private  

[5 rows x 65 columns]

In [7]:
len(data_b)

2435

In [8]:
data_w=data[data.race=='w']

In [9]:
data_w.head(5)

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
4  b  1          3       3        22       0          0         0         0   
5  b  1          4       2         6       1          0         0         0   
6  b  1          4       2         5       0          1         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   
5            266    ...          0.0     0.0    0.0       0.0       0.0   1.0   
6             13    ...          0.0     0.0    0.0       0.0       0.0   1.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  
5        0.0        0.0      0.0    Private  
6        0.0        0.0      0.0    Private  

[5 rows x 65 columns]

In [10]:
len(data_w)

2435

In [11]:
b_calls = data_b.call

In [12]:
w_calls = data_w.call

In [13]:
#mean of both samples and their difference - (0.09650924056768417, 0.0644763857126236, 0.03203285485506058)
w_calls.mean(),b_calls.mean(),w_calls.mean()-b_calls.mean()

(0.09650924056768417, 0.0644763857126236, 0.03203285485506058)

In [14]:
#standard deviation of both samples - (0.2953455150127411, 0.24564945697784424)
w_calls.std(),b_calls.std()

(0.2953455150127411, 0.24564945697784424)

In [18]:
standard_error = math.sqrt(w_calls.var()/len(w_calls) + b_calls.var()/len(b_calls))

In [19]:
standard_error * 1.960

0.015258417380692

confidence_interval = +- 0.01526, i.e. any difference above or below this interval will lead to the rejection of Ho. In this case the difference in mean between the two samples is 0.032, so Ho can be rejected

In [20]:
0.03203285485506058/ standard_error

4.1147383735462695

In [21]:
stats.ttest_ind(w_calls,b_calls)

Ttest_indResult(statistic=4.1147052908617514, pvalue=3.9408021031288859e-05)

The p value is 3.9408021031288859e-05

4. The difference between the call-back rate for a White sounding names, 9.65% is more than the 
call-back rate for Black sounding names, 6.45%. 
What are the chances that this is due to a random chance? 
Using the two sample t test, we observe that the possibility that this difference is due to chance is
low, it is 0.003 % . We can say with more than 99.9% confidence that, people with White sounding names
have a higher possibility of getting a call back when applying for a job.


5. From the statistical evidence, there is a strong indication that race plays an important role in 
determining job search success.